dao-factory/src/main/java/com/iluwatar/daofactory/PostgresOrderDAO.java

```diff
@@ -0,0 +1,243 @@
+        Statement stmt;
+
+        // Wrap each string attribute with a quotation for query
+        senderName = "'" + senderName + "'";
+        rcvrName = "'" + rcvrName + "'";
+        destination = "'" + destination + "'";
+
+        // Combine the query with the variable value
+        String sql = "INSERT INTO orders (sender_name,receiver_name,destination) "
+                + "VALUES (" + senderName + "," + rcvrName + "," + destination + ");";
+
+        try {
+            // Creating statement and executing query
+            stmt = con.createStatement();
+            stmt.executeUpdate(sql);
+
+            // Closing the connection
+            stmt.close();
+            con.close();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    /**
+     * A method to find an existing Order from the database.
+     * Ex. Find order that the senderName is John and receiverName is Hiro.
+     * Others attribute can be left as null.
+     *
+     * @param senderName The sender_name attribute of the Order
+     * @param rcvrName The receiver_name attribute of the Order
+     * @param destination The destination attribute of the Order
+     * @return An Order TO containing the attribute if found, and null otherwise
+     */
+    public Order findOrder(String senderName, String rcvrName, String destination) {
+        Order template = new Order();
+        Connection con = PostgresDAOFactory.createConnection();
+        Statement stmt;
+        ResultSet rs;
+        String sql = "SELECT * FROM orders WHERE (";
+        boolean firstCondition = true;
+
+        // Combining the query with the variable value depending on the
+        // attribute that has been passed to the function
+        if (senderName == null && rcvrName == null && destination == null) {
+            return template;
+        }
+        // Combining the senderName attribute
+        if (senderName != null && !senderName.equals("")) {
+            senderName = "'" + senderName + "'";
+            sql += "sender_name = " +senderName;
+            firstCondition = false;
+        }
+        // Combining the receiverName attribute
+        if (rcvrName != null && !rcvrName.equals("")) {
+            rcvrName = "'" + rcvrName + "'";
+            if (firstCondition) {
+                sql += "receiver_name = " + rcvrName;
+                firstCondition = false;
+            } else {
+                sql += " AND receiver_name = " + rcvrName;
+            }
+        }
+        // Combining the destination attribute
+        if (destination != null && !destination.equals("")) {
+            destination = "'" + destination + "'";
+            if (firstCondition) {
+                sql += "destination = " + destination;
+            } else {
+                sql += " AND destination = " + destination;
+            }
+        }
+        // Closing the sql statement
+        sql += ");";
```

Comment:
General note about this kind of SQL query building using string concatenation

- High risk of SQL injection if you accidentally concatenate user input to your SQL queries
- Difficult to avoid syntax errors in non-trivial cases, when dynamic SQL gets more complex

I would recommend using JPA criteria queries or jOOQ to make the code cleaner and safer.